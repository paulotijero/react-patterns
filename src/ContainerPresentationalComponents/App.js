import Visibility from "./containers/Visibility";

export default Visibility;
