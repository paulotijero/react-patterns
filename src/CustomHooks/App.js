import React from "react";

import useVisibility from "./Visibility";

function App() {
  const state = useVisibility();

  return (
    <>
      <h1>{state.toString()}</h1>
    </>
  );
}

export default App;
