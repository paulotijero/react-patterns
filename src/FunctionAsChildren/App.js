import React from "react";

import Visibility from "./Visibility";

function App() {
  return (
    <>
      <Visibility>{state => <h1>{state.toString()}</h1>}</Visibility>
    </>
  );
}

export default App;
