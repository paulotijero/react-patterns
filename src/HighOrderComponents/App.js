import React from "react";

import withVisibility from "./Visibility";

function App({ state }) {
  return (
    <>
      <h1>{state.toString()}</h1>
    </>
  );
}

export default withVisibility(App);
