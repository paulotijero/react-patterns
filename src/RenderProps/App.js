import React from "react";

import Visibility from "./Visibility";

function App() {
  return (
    <>
      <Visibility render={state => <h1>{state.toString()}</h1>} />
    </>
  );
}

export default App;
