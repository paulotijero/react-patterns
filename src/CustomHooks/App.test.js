import React from "react";
import { render } from "@testing-library/react";

import App from "./App";

test("Custom Hooks Patter", () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});
