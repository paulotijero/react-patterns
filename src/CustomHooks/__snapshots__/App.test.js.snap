// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Hook Patter 1`] = `
<DocumentFragment>
  <h1>
    [object Object]
  </h1>
</DocumentFragment>
`;
